src/main/webapp/go.js

```diff
@@ -299,19 +339,40 @@ function sendIndexToServlet(index) {
  * Disable lint check because proceed() is called from go.html.
  */
 function proceed() { //eslint-disable-line
-  destIndex++;
-  sendIndexToServlet(destIndex);
-  if (destIndex < huntArr.length) {
-    updateMessage(RIDDLE_DISPLAY, 'Riddle: ' + huntArr[destIndex].puzzle);
-    toggleHintButton(/* hide = */ false);
+  hunt.destIndex = hunt.destIndex + 1;
+  sendIndexToServlet(hunt.getDestIndex());
+  if (hunt.getDestIndex() < hunt.getNumItems()) {
+    updateMessage(RIDDLE_DISPLAY, 'Riddle: ' + hunt.getCurDestPuzzle());
+    delayHintButton();
   } else {
     updateRiddleToFinalMessage();
     toggleHintButton(/* hide = */ true);
   }
+  updateProgressBar();
   toggleProceedButton(/* hide = */ true);
   deleteMessage(SUBMIT_DISPLAY);
   deleteMessage(HINT_DISPLAY);
-  hintIndex = 0;
+  hunt.setHintIndex(0);
+}
+
+/**
+ * Update the progress bar to reflect how far the user is
+ * on the scavenger hunt.
+ */
+function updateProgressBar() {
+  const bar = document.getElementById(PROGRESS_DISPLAY);
+  const newWidth = (hunt.getDestIndex() / hunt.getNumItems()) * 100;
```

Comment:
hunt could also have a method called getProgress that calculates this percentage